### ISSUE TITLE
Feature request - Adding Rootstock RSKj fullnode to Umbrel

### ISSUE THREAD

**Issue by royalgarter** [getumbrel/umbrel#795]
Hi community,

Last week, I learned about RSK and their features and totally amazed. I was learning to add a RSKj docker to current [Umbrel Node via Docker](https://developers.rsk.co/rsk/node/install/), but it's not done yet. So a shower thought that do you guys think about adding RSKj to Umbrel in the future?

**Comment by Goro2030** [comment]
I upvote your idea. RSK is a great development!

I can help you setup the docker part of it if you guide me to where the RSKj containers are defined.

**Comment by Goro2030** [comment]
I've found it on their site :)

[](https://developers.rsk.co/rsk/node/install/docker/)

But I think it would be too heavy ( runs on Java ) for a Raspberry Pi ... you can try it though. To do so, once your Umbrel boots, connect to it using SSH, and run

`curl https://raw.githubusercontent.com/rsksmart/artifacts/master/Dockerfiles/RSK-Node/Dockerfile.MainNet --output Dockerfile.MainNet`

Then, you need to generate the image, by running:
(BTW: The image is not optimized AT ALL, RSK could do much better here ... i.e. it uses a full 700 MB Ubuntu image for it!)

`docker build -t mainnet -f Dockerfile.MainNet .`

Then you can run it:

`docker run -d --name mainnet-node-01  -p 4444:4444 -p 5050:5050 mainnet`

You can see if it's working using "docker ps" and see how much is consuming using "docker stats"

I didn't try to compile it, so if you do that, please let me know how it went. It's Java-based, so I guess it will compile on ARM.

**Comment by Goro2030** [comment]
Confirmed: RSK won't run on a PI, but it will run on Docker on a PC ( AMD64 ), so if you run it on a PC, you can give it a try! I compiled its image, and it's running ... taking up 1 GB of RAM and 25% of 2 cores on a Ryzen 7 👍

**Comment by AaronDewes** [comment]
> Confirmed: RSK won't run on a PI

Have you tried this?

> I can help you setup the docker part of it if you guide me to where the RSKj containers are defined.

https://github.com/getumbrel/umbrel/blob/master/apps/README.md

The images need to be on Docker Hub to be packaged for Umbrel

**Comment by Goro2030** [comment]
> > Confirmed: RSK won't run on a PI
> 
> Have you tried this?
> 
> > I can help you setup the docker part of it if you guide me to where the RSKj containers are defined.
> 
> https://github.com/getumbrel/umbrel/blob/master/apps/README.md
> 
> The images need to be on Docker Hub to be packaged for Umbrel

Thanks for the link @AaronDewes . Since this application does not run on ARM, there's no point in creating an APP for it at this point.

I'd close this ISSUE.

**Comment by AaronDewes** [comment]
>  Since this application does not run on ARM

Have you tried doing so? The Docker container should build fine

**Comment by Goro2030** [comment]
Yes, I tried and built the whole docker image. But the RSK people do not have an architectural candidate for ARM in their repository (ppa), so it doesn't finish building. Ence my statement that "it doesn't work on ARM". It works on Linux though.

**Comment by royalgarter** [comment]
> Yes, I tried and built the whole docker image. But the RSK people do not have an architectural candidate for ARM in their repository (ppa), so it doesn't finish building. Ence my statement that "it doesn't work on ARM". It works on Linux though.

Same on me. Tried to using prebuild RSK image but it said "it doesn't work on ARM". Maybe self built RSK from source will do. But it will take more time to try it.

**Comment by notleanbarba** [comment]
> Hi community,
> 
> Last week, I learned about RSK and their features and totally amazed. I was learning to add a RSKj docker to current [Umbrel Node via Docker](https://developers.rsk.co/rsk/node/install/), but it's not done yet. So a shower thought that do you guys think about adding RSKj to Umbrel in the future?

I guess umbrel should be developed based on bitcoin and not another sidechain.

**Comment by Goro2030** [comment]
I propose the maintainers close this "OPEN" issue, as it's not an Issue, but a feature request not aligned with Umbrel's purpose.

**Comment by lukechilds** [comment]
I think it's ok to keep this issue open. We're definitely open to the idea of someone packaging an RSK node for Umbrel.

**Comment by Goro2030** [comment]
OK, if that's what you want, let's do that. I've opened a feature request with my friends at RSK.

[https://github.com/rsksmart/rskj/issues/1565](https://github.com/rsksmart/rskj/issues/1565)
